google/cloud/forseti/services/server_config/service.py

```diff
@@ -26,6 +29,48 @@
 LOGGER = logger.get_logger(__name__)
 
 
+def _run(client):
+    """Runs Forseti.
+
+    Args:
+        client (ClientComposition): Forseti client object.
+
+    Returns:
+        ServerRunReply: the returned proto message.
+    """
+    today = datetime.now()
+    model_name = today.strftime('run_model_%Y%m%d_%H%M')
+    message = 'Forseti server run complete'
+
+    inventory_id = 0
+
+    for progress in client.inventory.create():
+        inventory_id = progress.id
+
+    model_reply = client.model.new_model(
+        'inventory',
+        model_name,
+        inventory_id,
+        False)
+
+    model_handle = model_reply.model.handle
+    model_status = model_reply.model.status
+
+    client.switch_model(model_handle)
+
+    if model_status in ['SUCCESS', 'PARTIAL_SUCCESS']:
+        for progress in client.scanner.run(scanner_name=None):
+            pass
```

Comment:
If we are just passing here, can we just do below, instead of looping?
```
client.scanner.run(scanner_name=None)
client.notifier.run(inventory_id, 0)
```

Reply:
That would certainly be cleaner. Because the gRPC reply to a run request is a stream, we need to consume the entire stream before proceeding.  Without the loop, the execution would continue without waiting for the scan to complete, for example.